### PR TITLE
New profile: irssi

### PR DESCRIPTION
**Pull request by gcb**
cross irssi off the wanted list https://github.com/netblue30/firejail/issues/1139

I think it will work, but I don't have how to test DCC, because well, like the entire universe i'm behind layers of NAT, CGNAT, etc... and nobody should use DCC anyway.

### CODE REVIEW

File: etc/profile-a-l/irssi.profile [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+protocol unix,inet,inet6
+seccomp
+# NOTE: comments here are things that can be improved, if you can spare the time.
+##seccomp.drop SYSCALLS (see syscalls.txt)
+##seccomp-error-action log (only for debugging seccomp issues)
+#shell none
+#tracelog
+disable-mnt
+##private-opt NAME
+#private-tmp
```
Comment (rusty-snake) [inline etc/profile-a-l/irssi.profile:50]: `private-dev`? `private-cache`?

File: etc/profile-a-l/irssi.profile [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+##writable-run-user
+##writable-var
+##writable-var-log
```
Comment (rusty-snake) [inline etc/profile-a-l/irssi.profile:53]:
```suggestion
```

File: etc/profile-a-l/irssi.profile [hunk 3]
```diff
@@ -0,0 +1,67 @@
+# Firejail profile for irssi
```
Comment (rusty-snake) [inline etc/profile-a-l/irssi.profile:1]: firecfg.config?

File: etc/profile-a-l/irssi.profile [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+noblacklist ${HOME}/.irssi/
+
+include disable-common.inc
+include disable-devel.inc
+include disable-exec.inc
+include disable-interpreters.inc
+include disable-proc.inc
+include disable-programs.inc
+include disable-shell.inc
+include disable-write-mnt.inc
```
Comment (rusty-snake) [inline etc/profile-a-l/irssi.profile:19]:
```suggestion
```

File: etc/profile-a-l/irssi.profile [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+# Description: IRC client
+# This file is overwritten after every install/update
+# Persistent local customizations
+include irssi.local
+
+# Persistent global definitions
+include globals.local
+
+noblacklist ${HOME}/.irssi/
+
```
Comment (rusty-snake) [inline etc/profile-a-l/irssi.profile:11]:
```suggestion

blacklist ${RUNUSER}/wayland-*

```

**Review by rusty-snake — DISMISSED** [review]

File: etc/profile-a-l/irssi.profile [hunk 6]
```diff
@@ -0,0 +1,67 @@
+# Firejail profile for irssi
+# Description: IRC client
+# This file is overwritten after every install/update
+# Persistent local customizations
+include irssi.local
+
+# Persistent global definitions
+include globals.local
+
+noblacklist ${HOME}/.irssi/
```
[the remark this answers is not in the document]
Comment (gcb) [reply etc/profile-a-l/irssi.profile]: habit of always adding this when allowing access later :) 

will drop

File: etc/profile-a-l/irssi.profile (hunk 1 above)
Comment (gcb) [reply etc/profile-a-l/irssi.profile:49]: lost those when moving my profile to the template. thanks.

File: etc/profile-a-l/irssi.profile [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+include disable-shell.inc
+include disable-write-mnt.inc
+include disable-X11.inc
+include disable-xdg.inc
+
+mkdir ${HOME}/.irssi/
+whitelist ${HOME}/.irssi/
+
+#include whitelist-usr-share-common.inc
+#include whitelist-var-common.inc
```
[the remark this answers is not in the document]
Comment (gcb) [reply etc/profile-a-l/irssi.profile]: yep. for some reason i was assuming they were subsets of the whitelist-common if you needed only those root dirs. :facepalm:

File: etc/profile-a-l/irssi.profile [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+include disable-exec.inc
+include disable-interpreters.inc
+include disable-proc.inc
+include disable-programs.inc
+include disable-shell.inc
+include disable-X11.inc
+include disable-xdg.inc
+
+mkdir ${HOME}/.irssi/
+whitelist ${HOME}/.irssi/
```
Comment (kmk3) [inline etc/profile-a-l/irssi.profile:22]: Remove trailing `/`, add `noblacklist`, add to disable-programs.inc.

File: etc/profile-a-l/irssi.profile [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
```
Comment (kmk3) [inline etc/profile-a-l/irssi.profile:23]:
```suggestion
```

File: etc/profile-a-l/irssi.profile [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+no3d
+nodvd
+nogroups
+nonewprivs
+noprinters
+noroot
+nosound
+notv
+nou2f
+novideo
```
Comment (kmk3) [inline etc/profile-a-l/irssi.profile:39]: Add and sort:

```
noinput
notpm
```

Note: `notpm` is in firejail-git.  When creating a new profile it's recommended
to use the latest development version to ensure that an up-to-date
profile.template is used.

File: etc/profile-a-l/irssi.profile [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+disable-mnt
+private-cache
+private-dev
+private-tmp
+
+dbus-user none
+dbus-system none
+
+# NOTE: almost sure irssi uses perl, but all seems to work without allowing it.
+
```
Comment (kmk3) [inline etc/profile-a-l/irssi.profile:53]:
```suggestion
```

**Review by kmk3 — CHANGES_REQUESTED** [review]

File: etc/profile-a-l/irssi.profile (hunk 10 above)
Comment (gcb) [reply etc/profile-a-l/irssi.profile:39]: I did use the template 4mo ago, when notpm was added to the template :)

Comment (gcb) [reply etc/profile-a-l/irssi.profile:39]: hum... i thought i had issues with noinput but now i tested and can't find why.  adding it back then

File: etc/profile-a-l/irssi.profile [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+notv
+nou2f
+novideo
+protocol unix,inet,inet6
+seccomp
+seccomp.block-secondary
+
+disable-mnt
+private-cache
+private-dev
```
Comment (kmk3) [inline etc/profile-a-l/irssi.profile:52]:
```suggestion
private-dev
private-etc @network,@tls-ca
```

Does it work with `private-etc`?

Comment (gcb) [reply etc/profile-a-l/irssi.profile:52]: I had installed -git to test with the `notpm` line, but now i'm back to my distro version and did not test `private-etc`. I will test with -git later on and report. But I don't see why not.

Comment (gcb) [reply etc/profile-a-l/irssi.profile:52]: instead I ran it now in debugger, also watching `lsof` for the pid, and nothing on etc is ever touched. I think it is OK.

Comment (kmk3) [reply etc/profile-a-l/irssi.profile:52]: > instead I ran it now in debugger, also watching `lsof` for the pid, and
> nothing on etc is ever touched. I think it is OK.

Can you test connecting to a server in firejail-git to make sure?

It seems strange that nothing on /etc would be accessed (such as TLS-related
paths).

What is the output of trace.txt in the following?

```sh
firejail --trace=trace.txt --profile=irssi /usr/bin/irssi
```

**Comment by netblue30** [comment]
Thanks for the patch, merging in!

**Comment by netblue30** [comment]
Added and commented out (see kmk3 above):
`````
# private-etc @network,@tls-ca
`````
@gcb give it a try, maybe is working. Thanks.

File: etc/profile-a-l/irssi.profile (hunk 12 above)
Comment (kmk3) [reply etc/profile-a-l/irssi.profile:52]: > My bad. My irssi profile is very uncommon. Retested this with a clean irssi
> profile, and indeed, it does break dns as expected.
> 
> with private-etc: Irssi: Unable to connect server irc.freenode.net port 6667
> [Temporary failure in name resolution]
> 
> without: resolves names fine.

Strange, what program do you use to manage dns?

On what distribution name/version?

Also, could you retest with the current irssi.profile in firejail-git (there is
one new change) and post the output of trace.txt?

**Pull request merged**